src/wallet/db.cpp

```diff
@@ -619,17 +637,7 @@ void BerkeleyEnvironment::Flush(bool fShutdown)
             } else
                 mi++;
         }
-        LogPrint(BCLog::WALLETDB, "BerkeleyEnvironment::Flush: Flush(%s)%s took %15dms\n", fShutdown ? "true" : "false", fDbEnvInit ? "" : " database not started", GetTimeMillis() - nStart);
-        if (fShutdown) {
```

Comment:
> Q: If there are multiple databases open in the same environment, previously logs would be consolidated and removed if any database was closed and no other database was writing, but now it will not consolidate logs until the last database in the environment is closed?

I don't think that's what happened previously either. `Flush(true)` seems to have been only called during a bitcoind shutdown, not during typical wallet unloading. So previously, even after a wallet was unloaded, it's log files were never consolidated and removed. So this behavior ends up being retained. I don't think there is a behavior change here, although the it is difficult to work out exactly when the database and environment are actually being flushed and closed.

Reply:
Also split this commit as you suggested.

Reply:
Done